### PR TITLE
Update to Cloud 2.0

### PR DESCRIPTION
**Pull request by calcastor**
This PR updates PGM to Cloud 2.0

This PR does not include general cleanup to take advantage of Cloud 2.0 improvements and changes, and is essentially a director port to start.

We removed command confirmation because it was never used at all from the original port over to Cloud.

### CODE REVIEW

File: core/src/main/java/tc/oc/pgm/command/parsers/MapPoolParser.java [hunk 1]
```diff
@@ -49,13 +49,12 @@ public final class MapPoolParser
...
   public @NotNull List<@NotNull String> stringSuggestions(
       final @NotNull CommandContext<CommandSender> context, final @NotNull CommandInput input) {
-    final String next = input.readString();
     MapOrder mapOrder = PGM.get().getMapOrder();
     if (!(mapOrder instanceof MapPoolManager)) return Collections.emptyList();
 
     return getPools(context, (MapPoolManager) mapOrder)
         .map(MapPool::getName)
-        .filter(name -> LiquidMetal.match(name, next))
+        .filter(name -> LiquidMetal.match(name, input.readString()))
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/command/parsers/MapPoolParser.java]: no, you can't do this.

File: core/src/main/java/tc/oc/pgm/command/parsers/PartyParser.java [hunk 2]
```diff
@@ -43,9 +43,8 @@ public ArgumentParseResult<Party> parse(
   @Override
   public @NotNull List<@NotNull String> stringSuggestions(
       @NotNull CommandContext<CommandSender> context, @NotNull CommandInput input) {
-    final String next = input.peekString();
     List<String> teams = teamParser.stringSuggestions(context, input);
-    if (LiquidMetal.match("obs", next)) teams.add("obs");
+    if (LiquidMetal.match("obs", input.peekString())) teams.add("obs");
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/command/parsers/PartyParser.java]: can't guarantee this either, you need to grab the state before it potentially gets manipulated by downstream

File: core/src/main/java/tc/oc/pgm/command/parsers/TeamsParser.java [hunk 3]
```diff
@@ -50,9 +50,8 @@ public ArgumentParseResult<Collection<Team>> parse(
   @Override
   public @NotNull List<@NotNull String> stringSuggestions(
       @NotNull CommandContext<CommandSender> context, @NotNull CommandInput input) {
-    final String next = input.peekString();
     List<String> teams = teamParser.stringSuggestions(context, input);
-    if ("*".startsWith(next)) teams.add("*");
+    if ("*".startsWith(input.peekString())) teams.add("*");
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/command/parsers/TeamsParser.java]: same here

File: core/src/main/java/tc/oc/pgm/command/parsers/VictoryConditionParser.java [hunk 4]
```diff
@@ -72,7 +71,7 @@ public final class VictoryConditionParser
             tmm.getParticipatingTeams().stream()
                 .map(Team::getNameLegacy)
                 .map(name -> name.replace(" ", "")))
-        .filter(str -> LiquidMetal.match(str, next))
+        .filter(str -> LiquidMetal.match(str, input.readString()))
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/command/parsers/VictoryConditionParser.java]: can't do this either

**Review by Pablete1234 — DISMISSED** [review]

**Comment by Pablete1234** [comment]
Let's hold off on merging this one purely to get other dependants ready

**Review by Pablete1234 — APPROVED** [review]

**Review by cswhite2000 — APPROVED** [review]

**Pull request merged**